src/tools/publish_toolstate.py

```diff
@@ -45,11 +56,48 @@ def read_current_status(current_commit, path):
                 return json.loads(status)
     return {}
 
+def issue(
+    tool,
+    maintainers,
+    relevant_pr_number,
+    relevant_pr_user,
+    pr_reviewer,
+):
+    # Open an issue about the toolstate failure.
+    gh_url = 'https://api.github.com/repos/rust-lang/rust/issues'
+    assignees = [x.strip() for x in maintainers.split('@') if x != '']
+    assignees.append(relevant_pr_user)
+    response = urllib2.urlopen(urllib2.Request(
+        gh_url,
+        json.dumps({
+            'body': textwrap.dedent('''\
+            Hello, this is your friendly neighborhood mergebot.
+            After merging PR {}, I observed that the tool {} no longer builds.
+            A follow-up PR to the repository {} is needed to fix the fallout.
+
+            cc @{}, do you think you would have time to do the follow-up work?
+            If so, that would be great!
+
+            cc @{}, the PR reviewer, and @rust-lang/compiler -- nominating for prioritization.
+
+            ''').format(relevant_pr_number, tool, REPOS[tool], relevant_pr_user, pr_reviewer),
```

Comment:
If we have a new enough version of Python, it might be more maintainable to use [named parameters](https://stackoverflow.com/questions/2451821/string-formatting-named-parameters) here, instead of positional. I don't have to touch this code, so it's no skin off my back!